Index: lxdm-0.5.3/data/themes/GalliumOS/gtkrc
===================================================================
--- lxdm-0.5.3.orig/data/themes/GalliumOS/gtkrc
+++ lxdm-0.5.3/data/themes/GalliumOS/gtkrc
@@ -16,13 +16,13 @@ style "back"
 
 style "time"
 {
-	fg[NORMAL]="#ffffff"
+	fg[NORMAL]="#cccccc"
 	font_name="Sans 12"
 }
 
 style "prompt"
 {
-	fg[NORMAL]="#000000"
+	fg[NORMAL]="#ffffff"
 	font_name="Sans Bold 14"
 }
 
Index: lxdm-0.5.3/data/themes/GalliumOS/gtk.css
===================================================================
--- lxdm-0.5.3.orig/data/themes/GalliumOS/gtk.css
+++ lxdm-0.5.3/data/themes/GalliumOS/gtk.css
@@ -9,12 +9,12 @@
 
 #time {
 	font: Sans 12;
-	color: #ffffff;
+	color: #cccccc;
 }
 
 #prompt {
 	font: Sans 14;
-	color: #000000;
+	color: #ffffff;
 }
 
 #bottom_pane GtkLabel {
